fuse/fuse_test.go

```diff
@@ -17,104 +19,290 @@ import (
 	if err != nil {
-		t.Fatalf("Cannot create mount: %v", err)
+		log.Fatalf("Cannot create mount: %v", err)
+		return nil, err
 	}
+	return mount, err
+}
+
+type brigPayload struct {
+	Path string
+	Data []byte
+}
+
+func (bmh *brigHelp) brigStage(ctx context.Context, req brigPayload) (*nothing, error) {
+	err := bfInfo.bfs.Stage(req.Path, bytes.NewReader(req.Data))
+	return &nothing{}, err
+}
 
-	f(mount)
+// Get data from a file stored by brig fs
+func (bmh *brigHelp) brigGetData(ctx context.Context, req brigPayload) (*brigPayload, error) {
+	out := brigPayload{}
+	out.Path = req.Path
 
-	if err := mount.Close(); err != nil {
-		t.Fatalf("Closing mount failed: %v", err)
+	stream, err := bfInfo.bfs.Cat(req.Path)
+	if err != nil {
+		log.Fatalf("Could not get stream for a brig file: %v", err)
+		return &out, err
 	}
+	result := bytes.NewBuffer(nil)
+	_, err = stream.WriteTo(result)
+	if err != nil {
+		log.Fatalf("Streaming to a buffer failed: %v", err)
+		return &out, err
+	}
+	out.Data = result.Bytes()
+
+	return &out, err
 }
 
-func withMount(t *testing.T, opts MountOptions, f func(mount *Mount)) {
-	withDummyFS(t, func(fs *catfs.FS) {
-		withMountFromFs(t, opts, fs, f)
-	})
+var helpers spawntest.Registry
+var brigHelper = helpers.Register("brigHelp", &brigHelp{})
 
+type mountInfo struct { // fuse related info available to OS layer
+	Dir  string
+	Opts MountOptions
 }
 
-func checkForCorrectFile(t *testing.T, path string, data []byte) {
-	// Try to read it over fuse:
-	helloBuffer := &bytes.Buffer{}
-	fd, err := os.Open(path)
+func withMount(t *testing.T, opts MountOptions, f func(ctx context.Context, control *spawntest.Control, mount *mountInfo)) {
+	// set up mounts
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	control := brigHelper.Spawn(ctx, t)
+	defer control.Close()
+
+	dbPath, err := ioutil.TempDir("", "brig-fs-test")
 	if err != nil {
-		t.Fatalf("Unable to open simple file over fuse: %v", err)
+		t.Fatalf("Failed to create temp dir for brig file system: %v", err)
 	}
+	require.Nil(t, err)
 
-	defer func() {
-		if err := fd.Close(); err != nil {
-			t.Fatalf("Unable to close simple file over fuse: %v", err)
-		}
-	}()
+	req := mountingRequest{
+		DbPath:  dbPath,
+		MntPath: filepath.Join(os.TempDir(), "brig-fuse-mountdir"),
+		Opts:    opts,
+	}
 
-	n, err := io.CopyBuffer(helloBuffer, fd, make([]byte, 128*1024))
-	if err != nil {
-		t.Fatalf("Unable to read full simple file over fuse: %v", err)
+	if err := control.JSON("/mount").Call(ctx, req, &nothing{}); err != nil {
+		t.Fatalf("calling helper: %v", err)
 	}
+	require.Nil(t, err)
 
-	if n != int64(len(data)) {
-		t.Fatalf("Data differs over fuse: got %d, should be %d bytes", n, len(data))
+	mount := mountInfo{Dir: req.MntPath, Opts: req.Opts}
+	// function which required mounts
+	f(ctx, control, &mount)
+
+	// cleanup and unmount
+	if err := control.JSON("/unmount").Call(ctx, nothing{}, &nothing{}); err != nil {
```

Comment:
It might be an idea to execute this code in a defer func, so it gets executed always, even if the test panics. Otherwise we might still have a (dysfunctional) mount lying around in that case.

Reply:
done in 34b7988